src/hooks/useMathProblems.ts

```diff
@@ -22,6 +30,18 @@ export default function useMathQuestions() {
     }
   }, [difficultyScore]);
 
+  const onTimeComplete = useCallback(
+    (route, navigation) => {
+      // FIXME: route and navigation type
+
+      statsMap.current.difficultyScore = difficultyScore;
+      statsMap.current.timePerQuestion =
+        TOTAL_TIME / statsMap.current.questionsAttemped;
+      navigation.navigate(...route.params.nextScreenArgs);
```

Comment:
Replace `navigation.navigate` with `navigation.replace` (This will unmount the `MathMain` screen when time runs outs, and prevent bugs related to pausing)